stacker/tests/test_plan.py

```diff
@@ -60,20 +60,47 @@ def test_build_plan(self):
             'namespace-bastion.1': set(['namespace-vpc.1']),
             'namespace-vpc.1': set([])})
 
+    def test_build_plan_missing_dependency(self):
+        bastion = Stack(
+            definition=generate_definition(
+                'bastion', 1, requires=['namespace-vpc.1']),
+            context=self.context)
+
+        plan = Plan(description="Test", sleep_func=None)
+
+        try:
+            plan.build([bastion])
+            self.assertFail()
+        except GraphError as e:
+            message = ("Error detected when adding 'namespace-vpc.1' "
+                       "as a dependency of 'namespace-bastion.1': node "
+                       "namespace-vpc.1 does not exist")
+            self.assertEqual(e.message, message)
+
     def test_build_plan_cyclic_dependencies(self):
         vpc = Stack(
             definition=generate_definition(
-                'vpc', 1, requires=['namespace-bastion.1']),
+                'vpc', 1),
             context=self.context)
-        bastion = Stack(
+        db = Stack(
             definition=generate_definition(
-                'bastion', 1, requires=['namespace-vpc.1']),
+                'db', 1, requires=['namespace-app.1']),
+            context=self.context)
+        app = Stack(
+            definition=generate_definition(
+                'app', 1, requires=['namespace-db.1']),
             context=self.context)
 
         plan = Plan(description="Test", sleep_func=None)
 
-        with self.assertRaises(CyclicDependencyError):
-            plan.build([vpc, bastion])
+        try:
```

Comment:
Oh sweet. I couldn't find an example online that used assertRaises to check the message. Much nicer.